File: samples/CSharpSamples/CSharpSamples/Samples/TextSample.cs
using Mku.Salmon.Text;

namespace Mku.Salmon.Samples.Samples;

class TextSample
{
    public static string EncryptText(string text, byte[] key)
    {
        // Always request a new random secure nonce.
        byte[] nonce = SalmonGenerator.GetSecureRandomBytes(8);

        // encrypt string and embed the nonce in the header
        string encText = SalmonTextEncryptor.EncryptString(text, key, nonce, true);
        return encText;
    }

    public static string DecryptText(string encText, byte[] key)
    {
        // decrypt string, the nonce is already embedded
        string decText = SalmonTextDecryptor.DecryptString(encText, key, null, true);
        return decText;
    }
}

File: samples/JavaSamples/src/main/java/com/mku/salmon/samples/samples/TextSample.java
package com.mku.salmon.samples.samples;

import com.mku.salmon.SalmonGenerator;
import com.mku.salmon.text.SalmonTextDecryptor;
import com.mku.salmon.text.SalmonTextEncryptor;

import java.io.IOException;


public class TextSample {
    public static String encryptText(String text, byte[] key) throws IOException {
        // Always request a new random secure nonce.
        byte[] nonce = SalmonGenerator.getSecureRandomBytes(8);

        // encrypt String and embed the nonce in the header
        String encText = SalmonTextEncryptor.encryptString(text, key, nonce, true);
        return encText;
    }

    public static String decryptText(String encText, byte[] key) throws IOException {
        // decrypt String, the nonce is already embedded
        String decText = SalmonTextDecryptor.decryptString(encText, key, null, true);
        return decText;
    }
}